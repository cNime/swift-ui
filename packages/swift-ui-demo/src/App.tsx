import { defineComponent, ref } from 'vue';
// import { SButton } from './components'
import { Button } from '/@/swift-ui/src/index.ts'
import '/@/swift-ui/src/style/components/loading.scss'
import '/@/swift-ui/src/style/components/accordion.scss'
import '/@/swift-ui/src/style/components/icon.scss'
import '/@/swift-ui/src/style/index.scss'
import Index from './index.vue'
export default defineComponent({
  name: 'App',
  setup() {
    const value1 = ref(false)
    return {
      value1
    };
  },
  methods: {
    openModal() {
      this.$modal({
        title: 'Open Modal as Method',
        'cancel-text': 'Cancel',
        'confirm-text': 'Confirm',
        content: <p>This is modal content This is modal content</p>
      })
    }
  },
  render() {
    const { openModal } = this
    return (
      <div id="nav">
        <Button onClick={openModal}>open Modal</Button>
        <Index />
      </div>
    )
  }
});