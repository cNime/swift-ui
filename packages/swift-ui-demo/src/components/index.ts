export { default as SButton } from './button1'