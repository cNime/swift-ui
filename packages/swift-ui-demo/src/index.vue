<template>
  <div>
    <ActionSheetPage />
  </div>
</template>