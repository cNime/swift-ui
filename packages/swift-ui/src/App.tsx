import { defineComponent } from 'vue';
import HelloWorld from './components/HelloWorld'
import SButton from './components/button'

export default defineComponent({
  name: 'App',
  setup() {
    return () => (
      <>
        <div id="nav">
          <HelloWorld></HelloWorld>
          <SButton>SButton</SButton>
        </div>
      </>
    );
  }
});