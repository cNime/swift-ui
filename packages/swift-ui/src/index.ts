import SwiftUi from './plugins'
import Loading from './components/loading/index'
import Accordion from './components/accordion/index'
import ActionSheet from './components/action-sheet/index'
import ActionSheetItem from './components/action-sheet/components/item'
import ActivityIndicator from './components/activity-indicator/index'
import Badge from './components/badge/index'
import Button from './components/button/index'
import Checkbox from './components/checkbox/index'
import List from './components/list/index'
import ListItem from './components/list/item/index'
import Divider from './components/divider/index'
import Drawer from './components/drawer/index'
import FloatLayout from './components/float-layout/index'
import Toast from './components/toast/index'
import Input from './components/input/index'
import Form from './components/form/index'
import Message from './components/message/index'
import Modal from './components/modal/index'
import ModalHeader from './components/modal/header/index'
import ModalAction from './components/modal/action/index'
import ModalContent from './components/modal/content/index'
import NavBar from './components/nav-bar/index'
import Noticebar from './components/noticebar/index'
import Pagination from './components/pagination/index'
import Progress from './components/progress/index'
import Radio from './components/radio/index'

export {
  Loading,
  Accordion,
  ActionSheet,
  ActionSheetItem,
  ActivityIndicator,
  Badge,
  Button,
  Checkbox,
  List,
  ListItem,
  Divider,
  Drawer,
  FloatLayout,
  Toast,
  Input,
  Form,
  Message,
  Modal,
  ModalHeader,
  ModalAction,
  ModalContent,
  NavBar,
  Noticebar,
  Pagination,
  Progress,
  Radio
}

export default SwiftUi
