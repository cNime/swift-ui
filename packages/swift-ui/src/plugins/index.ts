import { defineComponent } from 'vue'
import toast from './toast'
import message from './message'

export default {
  install: (app: ReturnType<typeof defineComponent>): void => {
    app.config.globalProperties.$toast = toast
    app.config.globalProperties.$message = message
  }
}
