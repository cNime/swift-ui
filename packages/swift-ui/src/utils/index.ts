const objectToString = (style: string | unknown): string => {
  if (style && typeof style === 'object') {
    let styleStr = ''
    Object.keys(style).forEach((key) => {
      const lowerCaseKey = key.replace(/([A-Z])/g, '-$1').toLowerCase()
      styleStr += `${lowerCaseKey}:${style[key]};`
    })
    return styleStr
  } else if (style && typeof style === 'string') {
    return style
  }
  return ''
}
export function mergeStyle(style1: string | unknown, style2: string | unknown): string | unknown {
  if (style1 && typeof style1 === 'object' && style2 && typeof style2 === 'object') {
    return Object.assign({}, style1, style2)
  }
  return objectToString(style1) + objectToString(style2)
}
