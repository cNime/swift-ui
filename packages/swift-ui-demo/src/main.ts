import { createApp } from 'vue'
import App from './App.vue'
import './app.scss';
import router from './router'
import '/@/swift-ui/src/style/index.scss'
import DocsHeader from './components/DocsHeader.vue'

const app = createApp(App)
app.component('DocsHeader', DocsHeader)
app.use(router).mount("#app");