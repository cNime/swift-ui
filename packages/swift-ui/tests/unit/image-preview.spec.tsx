import { mount } from '@vue/test-utils'
import { ImagePreview } from '../../src/index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(ImagePreview, {
    slots,
    props: { ...values },
  })
}

const files = [
  'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
  'https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg',
  'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
  'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
  'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
]
describe('ImagePreview Snap', () => {

  it('render initial ImagePreview', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ImagePreview props images', () => {
    const wrapper = factory({ images: files })
    expect(wrapper.element).toMatchSnapshot()
  })
})