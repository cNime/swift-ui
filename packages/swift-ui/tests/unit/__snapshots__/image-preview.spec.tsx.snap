// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePreview Snap render ImagePreview props images 1`] = `
<div
  class="at-image-preview"
>
  <div
    class="at-image-preview-mask"
  />
  <!---->
</div>
`;

exports[`ImagePreview Snap render ImagePreview props show 1`] = `
<div
  class="at-image-preview at-image-preview-active"
>
  <div
    class="at-image-preview-mask"
  />
  <div
    class="at-swiper horizontal at-image-preview-content"
    style="transform: scale(1, 1);"
  >
    <div
      class="at-swiper-wrap"
      style="transform: translate3d(0px,0px,0); transition-duration: 0ms; transition-timing-function: ease;"
    >
      
      
      <div
        class="at-swiper-slide"
      >
        <img
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
          style="max-width: 100%;"
        />
      </div>
      <div
        class="at-swiper-slide"
      >
        <img
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
          style="max-width: 100%;"
        />
      </div>
      <div
        class="at-swiper-slide"
      >
        <img
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
          style="max-width: 100%;"
        />
      </div>
      <div
        class="at-swiper-slide"
      >
        <img
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
          style="max-width: 100%;"
        />
      </div>
      <div
        class="at-swiper-slide"
      >
        <img
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
          style="max-width: 100%;"
        />
      </div>
      
      
    </div>
    <div
      class="at-swiper-pagination"
    >
      
      
    </div>
  </div>
</div>
`;

exports[`ImagePreview Snap render initial ImagePreview 1`] = `
<div
  class="at-image-preview"
>
  <div
    class="at-image-preview-mask"
  />
  <!---->
</div>
`;
